cmd/containerd-shim-runc-v2/process/init.go

```diff
@@ -276,6 +276,12 @@ func (p *Init) setExited(status int) {
 	p.exited = time.Now()
 	p.status = status
 	p.Platform.ShutdownConsole(context.Background(), p.console)
+	// We need to close the master tty here, in case of stdin,
```

Comment:
I think we can call `p.console.Close()` during deleting.

https://github.com/containerd/containerd/blob/a5aaa403b8a5aee1e98c5dcdf00cdb62b153b123/cmd/containerd-shim-runc-v2/process/init.go#L290-L292

Before deleting task, containerd always drains IO first. It's safe to close console in that place.

For the line 283 about calling `p.wg.Wait()`, I think it's unsafe to do that because `SetExited` is called by background goroutine triggered by SIGCHILD. When init process exits, its children might hold IO so that `p.wg.Wait()` will block until all its children exit. Check this issue https://github.com/containerd/containerd/issues/10094 for more details.

Reply:
Thanks @fuweid. That info really helped! I've made the suggested changes.

Reply:
@fuweid I think I finally root caused this issue. I updated the description of this PR with the info. Please take a look and lmk your thoughts.